web-client/app/scripts/services/linkService.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+angular.module('webClientApp')
+  .service('Link', ['$resource', '$http', '$q', 'API_HOST',
+      function ($resource, $http, $q, API_HOST) {
+
+      var baseUrl = '//' + API_HOST + '/api/v1/';
+      var LinkResource = $resource(baseUrl + 'links/:linkId');
+
+      /**
+       * These configs are needed. AngularJS identity tranformer
+       * can automatically figure out that this is a multipart
+       * content and use the correct Content-Type.
+       */
+      var configs = {
+        headers: {'Content-Type': undefined},
+        transformRequest: angular.identity
+      };
+
+      var createFormData = function (data) {
+        var fd = new FormData();
+        for (var key in data.link) {
+          if (key.trim() === '' || key.indexOf('$') === 0 || key === 'toJSON') {
+            continue;
+          }
+          fd.append('link[' + key + ']', data.link[key]);
+        }
+        return fd;
+      };
+
+      return {
```

Comment:
Yes, [AngularJS $resource](https://docs.angularjs.org/api/ngResource/service/$resource) adds all the cool stuff. See `topicService.js`